graylog2-server/src/main/java/org/graylog2/indexer/indices/Indices.java

```diff
@@ -589,19 +603,30 @@ public TimestampStats timestampStatsOfIndex(String index) {
         } catch (ElasticsearchException e) {
             LOG.error("Error while calculating timestamp stats in index <" + index + ">", e);
             throw new org.elasticsearch.index.IndexNotFoundException("Index " + index + " not found", e);
+        } catch (IOException e) {
+            // can potentially happen when recreating the source of
+            // the index range aggregation query on DEBUG (via XContentHelper)
+            throw new RuntimeException(e);
         }
 
         final Filter f = response.getAggregations().get("agg");
         if (f.getDocCount() == 0L) {
             LOG.debug("No documents with attribute \"timestamp\" found in index <{}>", index);
-            return TimestampStats.EMPTY;
+            return IndexRangeStats.EMPTY;
         }
 
         final Min minAgg = f.getAggregations().get("ts_min");
         final DateTime min = new DateTime((long) minAgg.getValue(), DateTimeZone.UTC);
         final Max maxAgg = f.getAggregations().get("ts_max");
         final DateTime max = new DateTime((long) maxAgg.getValue(), DateTimeZone.UTC);
+        List<String> streamIds = null;
+        final Terms streams = f.getAggregations().get("streams");
+        if (!streams.getBuckets().isEmpty()) {
+            streamIds = ImmutableList.copyOf(streams.getBuckets().stream()
+                    .map(Terms.Bucket::getKeyAsString)
+                    .collect(toSet()));
```

Comment:
Mongojack didn't work with Set, so I copy it into a list for storing.
It's only important that the list does not contain duplicates, which is achieved by the intermediate set.